File: app/src/main/java/org/zwsmith/myapplication/MyApplication.java
package org.zwsmith.myapplication;

import android.app.Application;
import android.content.Context;

import org.zwsmith.myapplication.core.dependencyInjection.AppComponent;
import org.zwsmith.myapplication.core.dependencyInjection.AppModule;
import org.zwsmith.myapplication.core.dependencyInjection.DaggerAppComponent;

public class MyApplication extends Application {
    private AppComponent appComponent;

    public static MyApplication get(Context context) {
        return (MyApplication) context.getApplicationContext();
    }

    @Override
    public void onCreate() {
        super.onCreate();
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/org/zwsmith/myapplication/MyApplication.kt
package org.zwsmith.myapplication

import android.app.Application

import org.zwsmith.myapplication.core.dependencyInjection.AppComponent
import org.zwsmith.myapplication.core.dependencyInjection.AppModule
import org.zwsmith.myapplication.core.dependencyInjection.DaggerAppComponent

class MyApplication : Application() {
    val appComponent: AppComponent by lazy {
        DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .build()
    }
}
